routes/builds.js

```diff
@@ -16,31 +16,29 @@ GET /builds?application=APP&branch=master # return all builds for APP and branch
 GET /builds?application=APP&latest # return latest build for APP
 GET /builds?application=APP&branch=master&latest # return latest builds for APP and branch master
 
-GET /builds?application=APP&latest&q=[docker|git|branch|timestamp] # return latest build info for APP
-GET /builds?application=APP&branch=master&latest&q=[docker|git|branch|timestamp] # return latest build info for APP and branch master
+GET /builds?application=APP&latest&part=[docker|git|branch|timestamp] # return latest build info for APP
+GET /builds?application=APP&branch=master&latest&part=[docker|git|branch|timestamp] # return latest build info for APP and branch master
 */
 router.get('/', function(req, res, next) {
  if (req && req.query && req.query.application){
     Application.findOne( {'name': req.query.application }, function (err, application) {
         if (application){
-          queryString={application: application._id }
+          var queryString={};
+          queryString.application=application._id
           if (req.query.branch) {
             queryString.gitBranch= req.query.branch;
           }
           Build.find (queryString)
           .sort({"created_at": -1})
           .exec(function (err, builds) {
             if (err) return next(err);
-            if (typeof req.query.latest !== 'undefined') {
-              // do not require any valute for "latest"
+            if (req.query.latest == "") {
+              // lastes is defined, only get lastest one
               if (builds.length==0) {
-                var res_json = {
-                    "reason": "build not found with criteria"
-                }
-                res.status(HTTPStatus.NOT_FOUND).json(res_json);
```

Comment:
We don't. We already at the build model and it will return an empty array when nothing is matched. This is consistent with /build/ endpoint where if no build is found , it will just return an empty array.
Having said that, we might want to do some validation on the query input.